isis/tests/FunctionalTestsCsminit.cpp

```diff
@@ -0,0 +1,342 @@
+  EXPECT_EQ(stateString.Type().toStdString(), "String");
+
+  // Check that the plugin can create a model from the state string
+  std::string modelName = QString(blobPvl.findKeyword("ModelName")).toStdString();
+  EXPECT_TRUE(plugin->canModelBeConstructedFromState(modelName, stateString.string()));
+
+  // check blob label ModelName and Plugin Name
+  EXPECT_EQ(QString(blobPvl.findKeyword("PluginName")).toStdString(), plugin->getPluginName());
+  EXPECT_EQ(QString(blobPvl.findKeyword("ModelName")).toStdString(), AlternativeTestCsmModel::SENSOR_MODEL_NAME);
+
+  // Check the CsmInfo group
+  ASSERT_TRUE(testCube->hasGroup("CsmInfo"));
+  PvlGroup &infoGroup = testCube->group("CsmInfo");
+  ASSERT_TRUE(infoGroup.hasKeyword("CSMPlatformID"));
+  EXPECT_EQ(infoGroup["CSMPlatformID"][0].toStdString(), altModel.getPlatformIdentifier());
+  ASSERT_TRUE(infoGroup.hasKeyword("CSMInstrumentId"));
+  EXPECT_EQ(infoGroup["CSMInstrumentId"][0].toStdString(), altModel.getSensorIdentifier());
+  ASSERT_TRUE(infoGroup.hasKeyword("ReferenceTime"));
+  EXPECT_EQ(infoGroup["ReferenceTime"][0].toStdString(), altModel.getReferenceDateAndTime());
+  ASSERT_TRUE(infoGroup.hasKeyword("ModelParameterNames"));
+  ASSERT_EQ(infoGroup["ModelParameterNames"].size(), 3);
+  EXPECT_EQ(infoGroup["ModelParameterNames"][0].toStdString(), AlternativeTestCsmModel::PARAM_NAMES[0]);
+  EXPECT_EQ(infoGroup["ModelParameterNames"][1].toStdString(), AlternativeTestCsmModel::PARAM_NAMES[1]);
+  EXPECT_EQ(infoGroup["ModelParameterNames"][2].toStdString(), AlternativeTestCsmModel::PARAM_NAMES[2]);
+  ASSERT_TRUE(infoGroup.hasKeyword("ModelParameterUnits"));
+  ASSERT_EQ(infoGroup["ModelParameterUnits"].size(), 3);
+  EXPECT_EQ(infoGroup["ModelParameterUnits"][0].toStdString(), AlternativeTestCsmModel::PARAM_UNITS[0]);
+  EXPECT_EQ(infoGroup["ModelParameterUnits"][1].toStdString(), AlternativeTestCsmModel::PARAM_UNITS[1]);
+  EXPECT_EQ(infoGroup["ModelParameterUnits"][2].toStdString(), AlternativeTestCsmModel::PARAM_UNITS[2]);
+  ASSERT_TRUE(infoGroup.hasKeyword("ModelParameterTypes"));
+  ASSERT_EQ(infoGroup["ModelParameterTypes"].size(), 3);
+  EXPECT_EQ(infoGroup["ModelParameterTypes"][0].toStdString(), "FICTITIOUS");
+  EXPECT_EQ(infoGroup["ModelParameterTypes"][1].toStdString(), "REAL");
+  EXPECT_EQ(infoGroup["ModelParameterTypes"][2].toStdString(), "FIXED");
+}
+
+
+TEST_F(CSMPluginFixture, CSMinitFails) {
+  // Create an ISD that will result in no successful models
+  json isd;
+  isd["name"] = "failing_isd";
+  isd["test_param_one"] = "value_one";
+  isd["test_param_does_not_exist"] = "failing_value";
+
+  isdPath = tempDir.path() + "/failing.json";
+  std::ofstream file(isdPath.toStdString());
+  file << isd;
+  file.flush();
+
+  QVector<QString> args = {
+    "from="+filename,
+    "isd="+isdPath};
+
+  UserInterface options(APP_XML, args);
+
+  // Expect a failure due to being unable to construct any model from the isd
+  EXPECT_ANY_THROW(csminit(options));
```

Comment:
Same as above -- should we be testing the specific error message, or is any throw good enough?

Reply:
Thanks, good idea, updating this!